cli/tools/upgrade.rs

```diff
@@ -210,30 +210,96 @@ pub fn check_for_upgrades(
 
   // Print a message if an update is available
   if let Some(upgrade_version) = update_checker.should_prompt() {
+    let auto_upgrade_env = match env::var("DENO_AUTO_UPGRADE") {
+      Ok(env_var) => env_var,
+      Err(_) => String::from(""),
+    };
     if log::log_enabled!(log::Level::Info) && atty::is(atty::Stream::Stderr) {
-      if version::is_canary() {
-        eprint!(
-          "{} ",
-          colors::green("A new canary release of Deno is available.")
-        );
-        eprintln!(
-          "{}",
-          colors::italic_gray("Run `deno upgrade --canary` to install it.")
-        );
+      if auto_upgrade_env != "1" {
+        if version::is_canary() {
+          eprint!(
+            "{} ",
+            colors::green("A new canary release of Deno is available.")
+          );
+          eprintln!(
+            "{}",
+            colors::italic_gray("Run `deno upgrade --canary` to install it.")
+          );
```

Comment:
We should add another log here informing that setting `DENO_AUTO_UPGRADE` env var will enable automatic updates